From 9c6dada6ef648ce2dd33d31863b132d26d99dd3b Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Tue, 27 Apr 2021 11:54:13 +0300
Subject: [PATCH] [SW-36011] gaudi: security: fix nic protection

Fix NIC RX page protection, so that 0xF60-0xF80
are not blocked, as this space is required to be
accessed from host.

Change-Id: Ibbe9be7fc75f36fc06cb04074ecd85320b323ad0
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index d2e9801..4974222 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1021,6 +1021,9 @@
 	}
 }
 
+/* registers 0xF60-0xF80 are allowed access */
+#define PHY_RX_PROTECT_HIGH	0xF60
+
 static void preboot_secure_nic_phy(void)
 {
 	u32 mask, port, prot_base, base = mmNIC0_PHY_BASE & ~HL_CFG_BASE;
@@ -1033,10 +1036,10 @@
 			PROT_BIT(mmNIC0_PHY_PHY_RX_CFG_3));
 		regmap_writel(mask, PRIV_PROT_REG(base));
 
-		/* offsets: 0x80 - 0xF80 */
+		/* offsets: 0x80 - 0xF60 */
 		prot_base = base + SNGL_REG_PROT_SIZE;
 		set_prot_bits_range_all(prot_base,
-					PROT_SIZE_MAX - (prot_base - base));
+					PHY_RX_PROTECT_HIGH - SNGL_REG_PROT_SIZE);
 		base += GAUDI_NICS_DIST;
 	}
 }
